### ISSUE TITLE
After upgrede to 4.8.0, omnisci can't work as normal

### ISSUE THREAD

**Issue by titawork** [omnisci/omniscidb#384]
Following SQLs work well on version before 4.8.0,
Upgrade to 4.8.0,
1. when execute  <creaate view VIEW1... as select ...>, 
then <creaate view VIEW2... as select ... from VIEW1 group by ...>
error happend, logs as following,
2019-08-29T04:15:54.781942 I 196070 BufferMgr.cpp:291 ALLOCATION slab of 8388608 pages (4294967296B) created in 0 ms CPU_MGR:0
2019-08-29T04:15:56.623940 F 196070 Calcite.cpp:481 Error occurred trying to communicate with calcite server, the error was: 'THRIFT_EAGAIN (timed out)', omnisci_server restart will be required
2019-08-29T04:15:56.821463 I 196070 MapDServer.cpp:935 Interrupt signal (6) received.

2. when execute such sql <creaate view ... as select ...>,
nmon indicate that
![image](https://user-images.githubusercontent.com/13470550/63923493-3e80e900-ca79-11e9-85a0-12a4c9d45c64.png)
why so much write operation?

and the CPUs are waiting for something,
![image](https://user-images.githubusercontent.com/13470550/63923686-8c95ec80-ca79-11e9-83ca-a6df2b420782.png)

those SQL running OK with version before 4.8.0.

It's very stranger! What happend?

**Comment by cdessanti** [comment]
It's waiting on disk writes. The error is occurring when you creta e the second view or when you run a query against the secondo view?

**Comment by titawork** [comment]
Yes, error happend when the second SQL <creaate view VIEW2... as select ... from VIEW1 group by ...> excuted.

**Comment by titawork** [comment]
Acutally, these two SQLs willl not write disk, right?

**Comment by alexbaden** [comment]
@titawork are you able to post some more of the query that you used, or does this happen with any query built on a view?

**Comment by cdessanti** [comment]
On ddl then. Maybe an infinite loop allocating a large chuck of memory and swapping or a big core written to disk?

**Comment by alexbaden** [comment]
Yeah, that’s my guess. We need to do some more digging, we upgraded calcite recently but I am pretty sure that was after 4.8. @titawork if you’re able to reproduce this consistently and could check against the master branch of this repo, we would appreciate it.

**Comment by titawork** [comment]
Here are those two SQLs, 
Yerstaday, I installed back to 4.7.0, those two SQLs ran OK.
Today, I upgraded to 4.8.0 again, issue happend.
By the way, where I can get install file of version 4.7.1? Thanks!

CREATE VIEW business_m AS SELECT
A.e_type,
A.m_type,
A.r_account,
A.s_code,
A.c_s_type,
A.e_kind,
A.s_no,
A.t_id,
A.b_id,
A.b_date,
A.b_time,
A.b_price,
A.s_price,
A.b_amount,
A.s_amount,
A.b_balance,
A.s_balance,
A.c_balance,
A.s_date,
A.r_code,
A.b_remark,
A.f_fare1,
A.f_fare2,
A.f_fare3,
A.f_fare4,
A.f_fare5,
A.f_farex,
B.b_no,
B.o_B_way as B_way,
B.r_time,
B.f_kind,
B.B_type,
B.B_no,
B.room_code,
B.f_account,
B.s_account,
B.B_prop,
B.order_id,
B.B_bs,
C.f_k_str,
C.d_model,
D.s_type,
D.s_unit
from A,
B,
C,
D,
E,
F
where
A.s_no=F.s_no and
A.B_bs =B.B_bs and
A.r_account=B.r_account and
A.s_code=B.s_code and
A.e_type=B.e_type and
B.f_account=C.f_account and
A.s_code=D.s_code and
E.b_no=C.b_no






CREATE VIEW business_view AS
select
r_account,
s_no,
o_id,
s_code,
b_no,
e_type,
e_bs,
s_unit,
max(business_m.b_date) as e_date,
sum(b_amount * b_price)/sum(b_amount) as b_price,
sum(b_amount) as b_amount,
sum(b_balance) as b_balance,
sum(c_balance) as c_balance,
sum(f_fare1) as fare1,
sum(f_fare2) as fare2,
sum(f_fare3) as fare3,
sum(f_fare4) as fare4,
sum(f_fare5) as fare5,
sum(f_farex) as farex,
max(f_kind) as fare_kind,
max(e_no) as e_no,
max(room_code) as room_code,
max(d_model) as d_model,
max(e_way) as e_way,
e_type,
s_type
from business_m
group by r_account,s_no,o_id,s_code,b_no,e_bs,e_type,s_unit,e_type,s_type

**Comment by titawork** [comment]
@alexbaden I'd like to try with the master repo, But where I can get package of the master branch of this repo. Because I don't have environment to compile MapD yet. By the way, If you can send a link to download version 4.7.1, it could be very good.

**Comment by cdessanti** [comment]
@alexbaden @titawork 
I tried to reproduce the issue, but I can get a crash and a core dump only if I create a join view on 4.7.0, then I try to create a group by view referring the first one in 4.8.

`2019-08-30T09:09:39.942354 F 92605 Calcite.cpp:481 Error occurred trying to communicate with calcite server, the error was: 'THRIFT_EAGAIN (timed out)', omnisci_server restart will be required
2019-08-30 09:09:40 WARN  TIOStreamTransport:close:112 - Error closing output stream.
java.net.SocketException: Socket closed
        at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:118)
        at java.net.SocketOutputStream.write(SocketOutputStream.java:155)
        at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:82)
        at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:140)
        at java.io.FilterOutputStream.close(FilterOutputStream.java:158)
        at org.apache.thrift.transport.TIOStreamTransport.close(TIOStreamTransport.java:110)
        at org.apache.thrift.transport.TSocket.close(TSocket.java:235)
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:303)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Aborted (core dumped)
`
If I ask a describe of the first view with the \d command of omnisql, i can create second view without crashes

**Comment by titawork** [comment]
Hi @cdessanti,
Actually, the two SQLs can work on 4.7.0, but can't work on 4.8.0 in my situation.

**Comment by titawork** [comment]
Hi,
Today, I installed 4.8.0 using tarball method.This issue gone. I don't know why.
 I updated by yum method before. It is very strange.

**Comment by randyzwitch** [comment]
It appears this issue was resolved by a re-install. If this continues to be an issue, please open a new issue
